### DISCUSSION TITLE
More generic slots that should eb available on all intents but would require more developement

### DISCUSSION THREAD

**Discussion by thrhymes** [OHF-Voice/intents#690] [_Ideas]
What i miss in Alexa/Echo and others is more built in complexity for intents...

***All*** Intents should have slots like "at", "on", "in", "for" and "until", or even "if", or "triggered by"
If the single intent then defines his own "at"-slot, that would overwrite the inherited class...

I understand these would trigger the impromptu creation of automations, that would have to expire and be deleted after completion..
But would that not be a real step forward?

- _"at time"_ ... "at 10:30" ... Do the thing at a certain time

- _"on day|week|weekend|date"_ ... Do the thing on a certain day ... can be combined with "at" or is done on the next occurence of the day at the time of invoking the command
 
- _"in timespan"_ ... Do the thing ..."in 1 hour, 10 minutes and 25 seconds"
 
- _"for duration then action"_... "for 3 minutes" ... Do the thing for a certain time, then invoke the opposite command... toggle should toggle back, if you turn something on, turn it off then 
-- "open the blinds for 30 minutes" ... opens the blinds and then closes them as "closing" is the opposite action
-- "open the blinds for 30 minutes then set to 30%" ... opens the blinds fully and after x minutes reduces them to 30%
 
- _"until time|condition"_ ... Do the thing until a certain condition is met ... 
-- "until tomorrow"
-- "until midnight"
-- "until the waterpump is turned on" 
-- "until the living room temperature is greater than 15°C"

- _"if condition and condition or condition"_ ... Do the thing if a condition is met
-- "open the blinds if Peter is at home and Thom is not at home"

- _"triggered by condition"_ ... do the Thing the next time a trigger-criteria is met "Switch the light on triggered by Sunset"

"Turn on the Radio on Monday at 6:30 for 15 Minutes if i am home".... This would be a Dream!!

**Comment by nagyrobi** [comment]
...would open a can of worms trying to prank you and set various stuff to happen in the future, without you even knowing why things happen...

**Reply by thrhymes** [reply]
A simple "Todo"-Card showing those to display on the Dash-board makes pranking impossible.... maybe with a button to stop/delete/delay ... oh, the things i am dreaming of ;)
